File: vbox-app/src/main/java/com/kedzie/vbox/api/IBIOSSettings.java
package com.kedzie.vbox.api;

import java.util.HashMap;
import java.util.Map;

import android.os.Parcel;
import android.os.Parcelable;

import com.kedzie.vbox.api.jaxb.BIOSBootMenuMode;
import com.kedzie.vbox.soap.Asyncronous;
import com.kedzie.vbox.soap.KSOAP;
import com.kedzie.vbox.soap.VBoxSvc;

@KSOAP
public interface IBIOSSettings extends IManagedObjectRef, Parcelable {
	
static final ClassLoader LOADER = IBIOSSettings.class.getClassLoader();
	
	public static final Parcelable.Creator<IBIOSSettings> CREATOR = new Parcelable.Creator<IBIOSSettings>() {
		public IBIOSSettings createFromParcel(Parcel in) {
			VBoxSvc vmgr =  in.readParcelable(LOADER);
			String id = in.readString();
			Map<String, Object> cache = new HashMap<String, Object>();
			in.readMap(cache, LOADER);
			return (IBIOSSettings) vmgr.getProxy(IBIOSSettings.class, id, cache); 
		}
		public IBIOSSettings[] newArray(int size) {  
			return new IBIOSSettings[size]; 
		}
	};

	@KSOAP(cacheable=true) public boolean getLogoFadeIn();
	@Asyncronous public void setLogoFadeIn(@KSOAP("logoFadeIn") boolean logoFadeIn);
	
	@KSOAP(cacheable=true) public boolean getLogoFadeOut();
	@Asyncronous @KSOAP public void setLogoFadeOut(@KSOAP("logoFadeOut") boolean logoFadeOut);
	
	@KSOAP(cacheable=true) public int getLogoDisplayTime();
	@Asyncronous @KSOAP public void setLogoDisplayTime(@KSOAP(type="unsignedInt", value="logoDisplayTime") int logoDisplayTime);
	
	@KSOAP(cacheable=true) public String getLogoImagePath();
	@Asyncronous @KSOAP public void setLogoImagePath(@KSOAP("logoImagePath") String logoImagePath);

	@KSOAP(cacheable=true) public int getTimeOffset();
	@Asyncronous @KSOAP public void setTimeOffset(@KSOAP(type="unsignedInt", value="timeOffset") boolean acpiEnabled);
	
	@KSOAP(cacheable=true) public Boolean getACPIEnabled();
	@Asyncronous @KSOAP public void setACPIEnabled(@KSOAP("ACPIEnabled") boolean acpiEnabled);
	
	@KSOAP(cacheable=true) public boolean getIOAPICEnabled();
	@Asyncronous @KSOAP public void setIOAPICEnabled(@KSOAP("IOAPICEnabled") boolean ioAPICEnabled);
	
	@KSOAP(cacheable=true) public BIOSBootMenuMode getBootMenuMode();
	@Asyncronous @KSOAP public void setBootMenuMode(@KSOAP("bootMenuMode") BIOSBootMenuMode bootMenuMode);
	
	@KSOAP(cacheable=true) public boolean getPXEDebugEnabled();
	@Asyncronous @KSOAP public void setPXEDebugEnabled(@KSOAP("PXEDebugEnabled") boolean pxeDebugEnabled);
}

File: vbox-app/src/main/java/com/kedzie/vbox/api/IBIOSSettings.kt
package com.kedzie.vbox.api

import android.os.Parcelable
import com.kedzie.vbox.api.jaxb.BIOSBootMenuMode
import com.kedzie.vbox.soap.Cacheable
import com.kedzie.vbox.soap.Ksoap
import com.kedzie.vbox.soap.KsoapProxy


@KsoapProxy
@Ksoap
interface IBIOSSettings: IManagedObjectRef, Parcelable {

    @Cacheable(value = "LogoFadeIn")
    suspend fun getLogoFadeIn(): Boolean

    suspend fun setLogoFadeIn(@Cacheable(value = "LogoFadeIn") logoFadeIn: Boolean)

    @Cacheable(value = "LogoFadeOut")
    suspend fun getLogoFadeOut(): Boolean

    suspend fun setLogoFadeOut(@Cacheable(value = "LogoFadeOut") logoFadeOut: Boolean)

    @Cacheable(value = "LogoDisplayTime")
    suspend fun getLogoDisplayTime(): Int

    suspend fun setLogoDisplayTime(@Cacheable(value = "LogoDisplayTime") @Ksoap(type = "unsignedInt") logoDisplayTime: Int)

    @Cacheable(value = "LogoImagePath")
    suspend fun getLogoImagePath(): String

    suspend fun setLogoImagePath(@Cacheable(value = "LogoImagePath") logoImagePath: String)

    @Cacheable(value = "TimeOffset")
    suspend fun getTimeOffset(): Int

    suspend fun setTimeOffset(@Cacheable(value = "TimeOffset") @Ksoap(type = "unsignedInt") timeOffset: Int)

    @Cacheable(value = "ACPIEnabled")
    suspend fun getACPIEnabled(): Boolean

    suspend fun setACPIEnabled(@Cacheable(value = "ACPIEnabled") ACPIEnabled: Boolean)

    @Cacheable(value = "IOAPICEnabled")
    suspend fun getIOAPICEnabled(): Boolean

    suspend fun setIOAPICEnabled(@Cacheable(value = "IOAPICEnabled") IOAPICEnabled: Boolean)

    @Cacheable(value = "BootMenuMode")
    suspend fun getBootMenuMode(): BIOSBootMenuMode

    suspend fun setBootMenuMode(@Cacheable(value = "BootMenuMode") bootMenuMode: BIOSBootMenuMode)

    @Cacheable(value = "PXEDebugEnabled")
    suspend fun getPXEDebugEnabled(): Boolean

    suspend fun setPXEDebugEnabled(@Cacheable(value = "PXEDebugEnabled") PXEDebugEnabled: Boolean)
}